### PR TITLE
Create user, auth views and routes

### PR DESCRIPTION
**Pull request by katesliang**
## Overview

Added routes for authenticating a user and getting basic user info 

## Changes Made

-Created new /authenticate/ and /me/ url
-Create views for authenticating and viewing user's own profile
-Made test case to ensure auth works

## Test Coverage

-Manual testing via Postman
-Created and passed test case for Google auth

## Next Steps 

Implement update and delete user

### CODE REVIEW

File: src/api/settings.py [hunk 1]
```diff
@@ -16,3 +16,4 @@
 
 AUTH_PASSWORD_SALT = environ["AUTH_PASSWORD_SALT"]
 ACCESS_TOKEN_AGE = 60 * 15
+GOOGLE_DEBUG = environ["GOOGLE_DEBUG"]
```
Comment (alanna-zhou) [inline src/api/settings.py]: wait is `environ.get` better

File: src/person/views.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            self.serializer_class(request.user).data, status.HTTP_200_OK
+        )
+
+
+class DeveloperView(generics.GenericAPIView):
+    serializer_class = AuthenticateSerializer
+    permission_classes = api_settings.ADMIN_PERMISSIONS
+
+    def get(self, request):
+        """Get all users"""
```
Comment (alanna-zhou) [inline src/person/views.py]: r u gonna be consistent with doc string periods? `Get all users.`

File: src/person/views.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        people = User.objects.all()
+        result = []
+        for person in people:
+            result.append(UserSerializer(person).data)
+        return success_response(result, status.HTTP_200_OK)
+
+    def post(self, request):
+        """Create test user or return access token for test user if user_id is provided"""
```
Comment (alanna-zhou) [inline src/person/views.py]: and here

File: src/person/views.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+class DeveloperView(generics.GenericAPIView):
+    serializer_class = AuthenticateSerializer
+    permission_classes = api_settings.ADMIN_PERMISSIONS
+
+    def get(self, request):
+        """Get all users"""
+        people = User.objects.all()
+        result = []
+        for person in people:
+            result.append(UserSerializer(person).data)
```
Comment (alanna-zhou) [inline src/person/views.py]:
```
users = [UserSerializer(person, many=True).data for person in User.objects.all())
```

Comment (alanna-zhou) [reply src/person/views.py]: also, you should probably call it `user` and not a person lol

File: src/person/views.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        return success_response(result, status.HTTP_200_OK)
```
Comment (alanna-zhou) [inline src/person/views.py]: is status not default 200?

**Review by alanna-zhou — CHANGES_REQUESTED** [review]
almost valid :)

File: src/person/views.py (hunk 2 above)
Comment (katesliang) [reply src/person/views.py]: wait im bad i meant to include developerview w the update user PR loooool but tank

File: src/api/tests.py [hunk 6]
```diff
@@ -1 +1,37 @@
...
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework.test import APIClient
+
+from .constants import TEST_EMAIL
+from .constants import TEST_FIRSTNAME
+from .constants import TEST_LASTNAME
+
+
+class TestAuthenticate(TestCase):
```
Comment (chalo2000) [inline src/api/tests.py:12]: love these test cases 😍

File: src/person/controllers/authenticate_controller.py [hunk 7]
```diff
@@ -35,17 +35,17 @@ def create_access_token(self, user):
         return token.key
 
     def create_user(self, user_data):
-        """Creates new user (Django auth) from `user_data`"""
-        return User.objects._create_user(**user_data)
+        """Creates new user (Django auth) from `user_data.`"""
```
Comment (chalo2000) [inline src/person/controllers/authenticate_controller.py]: can we put the period outside of the grave accents?

**Review by chalo2000 — APPROVED** [review]
LGTM

File: src/api/tests.py (hunk 6 above)
Comment (katesliang) [reply src/api/tests.py:12]: im rly original

**Review by noah-solomon — APPROVED** [review]
so valid

**Review by alanna-zhou — APPROVED** [review]
doge

**Pull request merged**